crates/datasources/src/excel/table.rs

```diff
@@ -0,0 +1,64 @@
+use std::borrow::Cow;
+use std::sync::Arc;
+
+use calamine::{open_workbook, Reader, Xlsx};
+use datafusion::datasource::streaming::StreamingTable;
+use datafusion::datasource::TableProvider;
+use datafusion::physical_plan::streaming::PartitionStream;
+
+use super::xlsx_sheet_value_to_record_batch;
+use crate::common::url::DatasourceUrl;
+use crate::excel;
+use crate::excel::errors::ExcelError;
+use crate::excel::stream::ExcelSheetPartitionStream;
+use crate::object_store::generic::GenericStoreAccess;
+
+pub async fn excel_streaming_table(
+    _store_access: GenericStoreAccess,
+    source_url: DatasourceUrl,
+    sheet_name: Option<&str>,
+    has_header: bool,
+    infer_schema_length: usize,
+) -> Result<Arc<dyn TableProvider>, ExcelError> {
+    let mut workbook: Xlsx<_> = match source_url {
+        DatasourceUrl::File(path) => open_workbook(path)?,
+        DatasourceUrl::Url(_) => todo!("remote files not supported yet"),
+    };
+
+    // for now, this only infers table data from only the first sheet in a workbook
+    // when read_excel gets vectorised to real columns this should be updated
+    let sheet = sheet_name.map(Cow::Borrowed).unwrap_or_else(|| {
```

Comment:
I'm not sure that we're getting much out of the `Cow` wrapping, but I suppose it's not hurting anyone here.